env_docker.go

```diff
@@ -269,12 +269,19 @@ func (e *DockerEnvironment) SharedDir() string {
 	return e.dir
 }
 
+const dockerCPUEnvName = "DOCKER_CPUS"
+
 func (e *DockerEnvironment) buildDockerRunArgs(name string, ports map[string]int, opts StartOptions) []string {
 	args := []string{"--rm", "--net=" + e.networkName, "--name=" + dockerNetworkContainerHost(e.networkName, name), "--hostname=" + name}
 
 	// Mount the docker env working directory into the container. It's shared across all containers to allow easier scenarios.
 	args = append(args, "-v", fmt.Sprintf("%s:%s:z", e.dir, e.dir))
 
+	dockerCPUs := os.Getenv(dockerCPUEnvName)
```

Comment:
Any reason why this needs to be an env var instead of an `EnvironmentOption`? 🙂

Reply:
Because it wouldn't require any code changes to run a test with limited CPU time and I could imagine that there are some platforms where such an option wouldn't be available. I can convert it to an `EnvironmentOption`, not a problem. I don't know what's the correct approach here :smile:

Reply:
Yeah, I thought it was something along those lines. I think it's fine as we have such a pattern with `E2E_TEMP_DIR` env var as well.

How about having both `EnvironmentOption`, and env var, and `EnvironmentOption` can override it? That way you can make certain tests permanently limited as well. And maybe we can name env var as `E2E_DOCKER_CPUS`? 🙂 

Not a blocker, we can iterate as well.

Reply:
Thanks for the review! I have changed the environment variable's name and added the ability to do the same via environment options. @saswatamcode PTAL when you have time :bow: